packages/engine/errors/7008.md

```diff
@@ -0,0 +1,26 @@
+---
+original: "Member '{0}' implicitly has an '{1}' type."
+excerpt: "I don't know what type this member is supposed to be. Your tsconfig file says I should throw an error here."
```

Comment:
What about something along the lines..
```
excerpt: "You haven't specified a type for class member '{0}'"
```
Or:
```
excerpt: "You need to specify a type for class member '{0}' (where I'm pointing at.)"
```

I understand the need to blame the type checker lol ("I don't know...") but maybe the excerpt could already explain the problem and provide a solution straight away?. It can be pretty straight forward to tell the user that they're just missing the type.

Regarding tsconfig part:
```
(...) Your tsconfig file says I should throw an error here.
```

imo, this is OK.. but it's not particularly helpful in the excerpt because there are dozens of compiler options in `tsconfig` so as a user I'd probably think by reading the current exceprt "_OK, so I need to fix my tsconfig?_" or "_what exactly is wrong with my tsconfig?_"

Then I'd be forced to open the full description to find out the solution which doesn't mention the tsconfig option at all which is responsible from causing this TS error diagnostic.

So I think we could mention tsconfig in the description (I believe it's `noImplicitAny`) if the user is not so much interested in solving this error but knowing the cause of it. Something along the lines:
````
I'm showing you this error because you have `noImplicitAny` enabled in your `tsconfig` file. This means a class member whose type is not defined is not by default `any` but I expect you to explicitly tell me the type of it.

```ts
class Foo {
  bar;      // Error: implicit any
  baz: any; // OK: explicit any
}
```

(... other examples you already mentioned )
````

What do you think?

Reply:
Regarding `member`. I can't truly think of another alternative because a member can be a property, method, or getter/setter so it seems `member` is intentionally used as a wide term to include all kind of property types which can only be know once the user has defined the type of the member.

Although, some diagnostics do refer to them as `properties` so maybe it's fine (`.. class property '${0}'..`)

Reply:
I like the idea of explaining the tsconfig stuff in the description at the bottom! That way, once the user understands what the rule is, they can turn it off if it's not a good fit for their project.

That excerpt sounds clearer, but starting with "You..." makes me feel like we're blaming the user. How about something like this?

```
A type hasn't been specified for for class member '{0}'.
```

I avoided using `{0}` because the [contributor guide](https://github.com/mattpocock/ts-error-translator/blob/main/CONTRIBUTING.md) recommended against it. Are there some cases where it's okay and other cases where it's not?

Thinking it over, I think I like "member" better than the alternatives. The [TypeScript class docs](https://www.typescriptlang.org/docs/handbook/2/classes.html) use "member" pretty consistently. Even though even though I find it to be a little jargon-ey, at least it's a term that someone could look up in the docs.

Thanks for all the thoughtful feedback!! :smile: